### PR TITLE
NO-JIRA: E2E: use strings.CutPrefix instead of trim function to get containerID

### PR DESCRIPTION
**Pull request by mrniranjan**
earlier when using strings.Trim , if the containerID contains any  characters from "cri-o://" it can remove these characters giving us invalid containerID

Instead of using Trim use the strings.CutPrefix method to fetch the containerID

### CODE REVIEW

File: test/e2e/performanceprofile/functests/utils/pods/pods.go [hunk 1]
```diff
@@ -207,7 +207,11 @@ func GetContainerIDByName(pod *corev1.Pod, containerName string) (string, error)
 	}
 	for _, containerStatus := range updatedPod.Status.ContainerStatuses {
 		if containerStatus.Name == containerName {
-			return strings.Trim(containerStatus.ContainerID, "cri-o://"), nil
+			containerID := containerStatus.ContainerID
+			// Split the container ID at the "://" delimiter, keeping the second part:
+			containerID = strings.SplitN(containerID, "://", 2)[1]
+			return containerID, nil
+
```
Comment (ffromani) [inline test/e2e/performanceprofile/functests/utils/pods/pods.go:214]: this works, but it seems a textbook case of https://pkg.go.dev/strings#CutPrefix . Can we use `CutPrefix`?

Comment (jmencak) [reply test/e2e/performanceprofile/functests/utils/pods/pods.go:214]: good point
/lgtm cancel

Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/utils/pods/pods.go:214]: If the prefix is not found that also means it's junk , as we are  checking object of v1.ContainerStatus where it's of   '<type>://<container_id>'.

Comment (ffromani) [reply test/e2e/performanceprofile/functests/utils/pods/pods.go:214]: fair enough! I still think `CutPrefix` can help; but even if it is not, and we need to use `Split` or `SplitN`, we need a bit more robustness check

Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/utils/pods/pods.go:214]: updating the patch to use CutPrefix

**Review by MarSik — APPROVED** [review]
/lgtm

**Review by ffromani — COMMENTED** [review]
/lgtm

**Pull request merged**